docs/user/rstudio/ide/guide/data/connection-snippets.qmd

```diff
@@ -0,0 +1,88 @@
+---
+title: "Connection Snippets"
+---
+
+RStudio connections can be extended through two methods:
+
+-  **Snippet Files** on a server-based version of RStudio that provide support for new connections using R templates that enable users and system administrators to list and create new connections.
+
+-  **Packages** that do one or more of the following:
+
+    -   Implement the [Connections Contract](connection-contracts.html) to manage and explore data in the **Connections** Pane and/or
+
+    -   Supply Snippet Files or a Shiny Application to extend the **New Connection** dialog.
+
+## Snippet files
+
+A Connection snippet file is an R code snippet with additional metadata which is intended to initialize a connection. This file can be as simple as:
+
+    library(readr)
+    data <- read_csv(readr_example("mtcars.csv"))
+
+This snippet can be saved on a server-based version of RStudio under `/etc/rstudio/connections/`. Once that file is created and RStudio is restarted, RStudio will make this connection available as *Motor Trend Cars* underneath **Connect to Existing Data Sources**:
+
```

Comment:
How do you format the Connection snippet file. Is it just like a regular snippet? So save the code below in `/etc/rstudio/connections/r.snippet`?

```
snippet Motor Trend Cars
    library(readr)
    data <- read_csv(readr_example("mtcars.csv"))
```

Reply:
looked at the sparklyr package and there's a snippet for [Livy.R](https://github.com/sparklyr/sparklyr/blob/main/inst/rstudio/connections/Livy.R) that is helpful.

so the file would be `/etc/rstudio/connections/MotorTrendCars.R` (idk how you could get the spaces in between the words to show up in the wizard though) and it would contain:

```
library(readr)
data <- read_csv(readr_example("mtcars.csv"))
```

would be helpful to explicitly state that you need to save off a `.R` file in that directory.

Reply:
It only works on server-based versions, so I've added context and a link to the R packages section for distributing snippets on desktop. THanks!